### PR TITLE
Create `count_distinct` for DocSet

### PR DESCRIPTION
**Pull request by tranade**
- DocSet count now takes an optional `field` param, which is used for count unique

### CODE REVIEW

File: lib/sycamore/sycamore/docset.py [hunk 1]
```diff
@@ -115,13 +115,19 @@ def _truncate(s):
 
             pprint.pp(document, stream=stream)
 
-    def count(self, include_metadata=False, **kwargs) -> int:
+    def count(self, field: Optional[str] = None, include_metadata=False, **kwargs) -> int:
```
Comment (baitsguy) [inline lib/sycamore/sycamore/docset.py:118]: Can you move the field arg after include_metadata?

that way if someone is currently doing `ds.count(True)` they won't break

**Review by baitsguy — COMMENTED** [review]
Code looks good, will discuss offline if adding a `.count_distinct(key)` is a better approach

File: lib/sycamore/sycamore/docset.py (hunk 1 above)
Comment (tranade) [reply lib/sycamore/sycamore/docset.py:118]: working on making a separate `count_distinct` instead

**Review by tranade — COMMENTED** [review]
Previous commit also makes a separate count distinct

**Review by baitsguy — APPROVED** [review]

**Pull request merged**